server/app/mutations/grid_services/common.rb

```diff
@@ -84,6 +86,30 @@ def build_grid_service_secrets(existing_secrets)
       service_secrets
     end
 
+    def build_service_volumes()
+      service_volumes = []
+      self.volumes.each do |vol|
+        vol_spec = parse_volume(vol)
+        if vol_spec[:volume]
+          # Named volume, try to find the proper mapping for external volumes
+          # NULL stack services don't have revs, for those just use the "global" name
+          volume_name = vol_spec[:volume]
+          if self.stack.latest_rev
```

Comment:
The other `GridServices::Common` helper methods take `grid` and `stack` as arguments.

This could take a `{ volume: Volume }` mapping as an argument that passed to the `Services` mutations by the `Stacks` mutations? For plain services where this is run directly from the grid services API without any such volume mappings, this would then interpret the named volumes as literal `Volume.name`s.